packages/app/src/components/Visualization/Visualization.js

```diff
@@ -42,7 +42,7 @@ export class Visualization extends Component {
     onError = response => {
         let error
         if (response) {
-            switch (response.errorCode) {
+            switch (response.details.errorCode) {
```

Comment:
```suggestion
            switch (response.details?.errorCode) {
```

Reply:
can we always expect `details` or should we `?` it?

Reply:
My understanding looking at the dataEngine code is that the full api response always ends up in `details` for response statuses outside the 200-399 range.
See:
https://github.com/dhis2/app-runtime/blob/master/services/data/src/links/RestAPILink/fetchData.ts#L17
https://github.com/dhis2/app-runtime/blob/master/services/data/src/links/RestAPILink/fetchData.ts#L40

Adding the check does not harm tho.